File: 946c.py
s = list(input())
found = False
counter = 'a'
for i in range(len(s)):
    c = s[i]
    if c <= counter:
        s[i] = counter
        counter = chr(ord(counter)+1)
        if counter > 'z':
            print(''.join(s))
            found = True
            break

if not found:
    print(-1)


        

File: 946c.cpp
#include <string>
#include <iostream>

using namespace std;

int main()  {
    string s;
    cin >> s;
    char counter = 'a';
    bool found = false;
    for (int i = 0; i < s.length(); ++i)    {
	char c = s[i];
	if (c <= counter)   {
	    s[i] = counter;
	    counter += 1;
	    if (counter > 'z')	{
		cout << s;
		found = true;
		break;
	    }
    }
    return 0;
}
